src/main/java/org/apache/mesos/scheduler/recovery/RecoveryOfferRequirementProvider.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.mesos.scheduler.recovery;
+
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.offer.OfferRequirement;
+
+import java.util.List;
+
+/**
+ * Implementations of this interface provide OfferRequirement objects when presented with Tasks which have failed
+ * permanently or transiently.
```

Comment:
What are the OfferRequirements intended to be used for? to re-launch the failed tasks? (cant hurt to list/specify what the expected responses are here)